pkg/provisioning/service.go

```diff
@@ -168,19 +167,51 @@ func (s *Service) Delete(ctx context.Context, id string) error {
 	return nil
 }
 
-// getYamlFiles recursively reads folders in the path and collects paths to all
-// files that end with .yml or .yaml.
+// getYamlFiles reads files in the path and collects paths to all
+// files that end with .yml or .yaml. It only reads files in the top-level
+// directory and skips hidden files and symlinks.
 func (s *Service) getYamlFiles(path string) ([]string, error) {
-	var files []string
-	err := filepath.WalkDir(path, func(path string, fileInfo fs.DirEntry, err error) error {
-		if strings.HasSuffix(path, ".yml") || strings.HasSuffix(path, ".yaml") {
-			files = append(files, path)
-		}
-		return nil
-	})
+	if path == "" {
```

Comment:
We should probably fail here, an empty string doesn't really make sense as a path.